### PR TITLE
Introduces a multi-filter for trades/deposits and withdrawals

### LINKED ISSUE #2847: Introduce searching/filtering for deposits & withdrawals and trades
## Abstract

I have thousands transactions aggregated from multiple exchanges and i need to find one with particular amount, date, asset but no search available to achieve that. Proposing to add 200, 300, 400, 500 amounts or even custom to pull all transactions on one screen. Currently i have to go on each page and search it.

Screen Shot 2021-05-05 at 6 55 12 PM

### PR DESCRIPTION
**Pull request by kelsos**
Closes #2847

## Checklist

- [x] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

**Comment by kelsos** [comment]
## Some pending stuff
- [ ] Allow arrow down/up to select along with tab the next/previous suggestion when they exist
- [ ] Make the available filters also selectable via keyboard (tab/arrow)
- [ ] Integrate with deposits and withdrawals
- [ ] Integrate location filter.

**Review by LefterisJP — COMMENTED** [review]
Some very simple remarks after checking only some DEX trades. Did not check deposits/withdrawals yet but I assume is same.

### Move location filter in there too


Since location is also a filter it would also make sense imo to have it there with all the rest of the filters

### Date filtering should be a range



I tested it and it seems to only accept a certain date. Is there any way to add a range? If yes it should also be in the tooltip where date: appears.

### Pressing Enter




After having typed the filters I would expect pressing Enter to close the filter dropdown but it does not. If you press somewhere else it goes away, but pressing enter doing the same thing would be addition to the UX.

### Mention filters are ANDed

I guess when you give more than one filter they are all ANDed. Mention that in the filter dropdown explanation help.


----

These are some things I saw with a very quick look.

Also as you have said many times in the past, all those filters should also just go to the DB. I could see noticeable lag at an account with many many events.

File: docs/changelog.rst [hunk 1]
```diff
@@ -2,6 +2,7 @@
 Changelog
 =========
 
+* :feature:`2847` Users will now be able to use a combination of filter for trades and deposits/withdrawals.
```
Comment (LefterisJP) [inline docs/changelog.rst:5]:
```suggestion
* :feature:`2847` Users will now be able to use a combination of filters in the search for trades and deposits/withdrawals.
```

**Review by LefterisJP — APPROVED** [review]
LGTM, seems to work and the comments I had addressed.

> Make the available filters also selectable via keyboard (tab/arrow)

So you mean pressing down arrow when suggestions appear to select them?

**Review by LefterisJP — COMMENTED** [review]
Guess would also make sense to update the usage guide with a screenshot

**Comment by kelsos** [comment]
Yes give me a moment

**Pull request merged**